### PR TITLE
HPCC-26845 Fix problems with node_modules directories on windows

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by ghalliday** [comment]
This is actually 3 different fixes - I will add comments on each

@GordonSmith FYI

File: ecl/hql/hqlcollect.cpp [hunk 1]
```diff
@@ -517,9 +517,12 @@ void FileSystemDirectory::processDependencies(IDirectoryIterator * dir)
         const char * tail = pathTail(filename);
         if (file.isDirectory() == fileBool::foundYes)
         {
-            //If there is a directory within node_module, then treat that directory as a package
-            PackageDependency * depend = new PackageDependency(createIdAtom(tail), filename, false);
-            contents.append(*depend);
+            if (isValidIdentifier(tail))
```
Comment (ghalliday) [inline ecl/hql/hqlcollect.cpp:520]: This is an optimization - ignore any directories which cannot be modules (e.g. .bin, @xxx)

File: ecl/hql/hqlrepository.cpp [hunk 2]
```diff
@@ -36,6 +36,11 @@ static const char * queryExtractFilename(const char * urn)
     case '/':
         return urn;
     }
+#ifdef _WIN32
```
Comment (ghalliday) [inline ecl/hql/hqlrepository.cpp:39]: This is the fix for the issue.  It wasn't spotting c:\ as an absolute filename.

File: system/jlib/jdebug.cpp [hunk 3]
```diff
@@ -1096,7 +1096,11 @@ void getCpuInfo(unsigned &numCPUs, unsigned &CPUSpeed)
 static unsigned evalAffinityCpus()
 {
     unsigned numCpus = 0;
+#ifdef __64BIT__
+    DWORD64 ProcessAffinityMask, SystemAffinityMask;
```
Comment (ghalliday) [inline system/jlib/jdebug.cpp:1100]: This is causing a memory corruption in 64bit - which was causing a very strange crash on exit (and probably a long pause on release builds).  Should possibly be backported even further.

**Comment by ghalliday** [comment]
@shamser @richardkchapman please take a look.

File: system/jlib/jdebug.cpp (hunk 3 above)
Comment (richardkchapman) [reply system/jlib/jdebug.cpp:1100]: If you want it backported further (and probably even if you don't) then it should be in a separate JIRA/PR

**Review by richardkchapman — CHANGES_REQUESTED** [review]
Should be split in two

**Review by shamser — COMMENTED** [review]
@ghalliday Other than issues reported by Richard, everything else looks good.

**Review by richardkchapman — APPROVED** [review]

**Pull request merged**